packages/flutter_tools/lib/src/commands/clean.dart

```diff
@@ -28,38 +32,71 @@ class CleanCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    // Clean Xcode to remove intermediate DerivedData artifacts.
+    // Do this before removing ephemeral directory, which would delete the xcworkspace.
+    final FlutterProject flutterProject = FlutterProject.current();
+    if (xcode.isInstalledAndMeetsVersionCheck) {
```

Comment:
I don't think this check can handle being run on windows host machines, you might need a platform guard as well.

Reply:
Maybe `getCurrentHostPlatform() == HostPlatform.darwin_x64` needs to be added to isInstalledAndMeetsVersionCheck instead?

Reply:
Sounds reasonable. You can use `platform.isMacOS` for this check. It is an injected value so that is still testable